core/trino-main/src/test/java/io/trino/sql/planner/iterative/rule/TestApplyTableScanRedirection.java

```diff
@@ -164,7 +164,8 @@ public void testMismatchedTypes()
                     .execute(MOCK_SESSION, session -> {
                         assertThatThrownBy(() -> runner.createPlan(session, "SELECT source_col_a FROM test_table", WarningCollector.NOOP))
                                 .isInstanceOf(TrinoException.class)
-                                .hasMessageMatching("Redirected column mock_catalog.target_schema.target_table.destination_col_c has type bigint, different from source column .*MockConnectorTableHandle.*source_col_a.* type: varchar");
+                                .hasMessageMatching("Redirected column mock_catalog.target_schema.target_table.destination_col_c has type bigint, " +
+                                        "different from source column mock_catalog.test_schema.test_table.*source_col_a.* type: varchar");
```

Comment:
ideally be a bit more precise than `.*source_col_a.*`, so that the test "sees" we output column handle toString

and also include the
```
// TODO report source column name instead of ConnectorColumnHandle toString
```

here

Reply:
done